lib/rules/no-obscure-array-access.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const Rule = require('./base');
+
+function ERROR_MESSAGE(expression) {
+  return `Obscure expressions are prohibited. ${expression}. Please use Ember's get helper instead. i.e {{get list '0'}} `;
+}
+
+module.exports = class NoObscureArrayAccess extends Rule {
+  visitor() {
+    return {
+      PathExpression(node) {
+        // Regex test for any word character plus digit. i.e.  "test.1"
+        let regex = /\w+.\d/;
+        if (node.original && node.original.match(regex)) {
```

Comment:
Instead of checking original here, I'd prefer we look at `path.parts` and see if any are numbers.

Something like:

```js
let hasObscureAccess = path.parts.some(part => Number.isInteger(Number(part)))
if (hasObscureAccess) {
  this.log(...);
}
``